component/file_cache/file_cache.go

```diff
@@ -1226,12 +1226,27 @@ func (fc *FileCache) FlushFile(options internal.FlushFileOptions) error {
 		// Write to storage
 		// Create a new handle for the SDK to use to upload (read local file)
 		// The local handle can still be used for read and write.
+		var orgMode fs.FileMode
+		modeChanged := false
+
 		uploadHandle, err := os.Open(localPath)
 		if err != nil {
-			log.Err("FileCache::FlushFile : error [unable to open upload handle] %s [%s]", options.Handle.Path, err.Error())
-			return nil
-		}
+			if os.IsPermission(err) {
+				info, _ := os.Stat(localPath)
+				orgMode = info.Mode()
+				err = os.Chmod(localPath, os.FileMode(0666))
```

Comment:
Changing the file mode to 0666 will make the in progress read and write requests to success without respecting their original filemode, we are not maintaining any lock in flush, read, write.

Reply:
We do not want to lock because earlier also parallel operations on file was allowd so adding a new lock here may break some existing scenarios. Adding mode to read the file is required otherwise SDK will not be able to read the file and upload the data. There is a small window where permissions are granted even when user did not give it. Is there any other better solution here ?

Reply:
Instead of setting it to 0666 blindly, we shall just add read bits to existing mode here.